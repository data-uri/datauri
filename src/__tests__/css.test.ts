import path from 'path';
import DataURICSSParser from '../datauri/css';

const fixture = path.resolve(__dirname, './fixtures/fixture.gif');

describe('Data-uri CSS Parser', () => {
  it('should create a class with datauri background using target file name', async () => {
    expect(await DataURICSSParser(fixture)).toMatchSnapshot();
  });

  it('should create a class with datauri background using a defined name', async () => {
    expect(await DataURICSSParser(fixture, { className: 'foobar' })).toMatchSnapshot();
  });

  it('should create a class with datauri background with width', async () => {
    expect(await DataURICSSParser(fixture, { width: true })).toMatchSnapshot();
  });

  it('should create a class with datauri background with height', async () => {
    expect(await DataURICSSParser(fixture, { height: true })).toMatchSnapshot();
  });

  it('should create a class with datauri background with both width and height', async () => {
    expect(await DataURICSSParser(fixture, { width: true, height: true })).toMatchSnapshot();
  });

  it('should create a class with datauri background with background-size', async () => {
    expect(await DataURICSSParser(fixture, { backgroundSize: true })).toMatchSnapshot();
  });
});
