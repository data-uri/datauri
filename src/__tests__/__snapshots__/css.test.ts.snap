// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data-uri CSS Parser should create a class with datauri background using a defined name 1`] = `
"
.foobar {
  background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7');
}"
`;

exports[`Data-uri CSS Parser should create a class with datauri background using target file name 1`] = `
"
.fixture {
  background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7');
}"
`;

exports[`Data-uri CSS Parser should create a class with datauri background with background-size 1`] = `
"
.fixture {
  background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7');
  background-size: 1px 1px;
}"
`;

exports[`Data-uri CSS Parser should create a class with datauri background with both width and height 1`] = `
"
.fixture {
  background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7');
  width: 1px;
  height: 1px;
}"
`;

exports[`Data-uri CSS Parser should create a class with datauri background with height 1`] = `
"
.fixture {
  background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7');
  height: 1px;
}"
`;

exports[`Data-uri CSS Parser should create a class with datauri background with width 1`] = `
"
.fixture {
  background-image: url('data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7');
  width: 1px;
}"
`;
